src/gauss_pdf_and_cdf.py

```diff
@@ -0,0 +1,106 @@
+import numpy as np
+import plotly.graph_objects as go
+from sympy import Symbol, pi, E, sqrt, integrate, oo
+from abc import ABC, abstractmethod
+from scipy import stats
+import math
+
+
+class ProbabilisticDistribution(ABC):
+
+    @abstractmethod
+    def __init__(self):
+        pass
+
+    @abstractmethod
+    def save_figure(self):
+        pass
+
+
+class Visualizer:
+
+    @staticmethod
+    def visualize(x, y, title):
+        fig = go.Figure()
+        fig.add_trace(go.Scatter(x=x, y=y, name=title))
+        fig.update_xaxes(title="x")
+        fig.update_yaxes(title="y")
+        fig.update_layout(title=title)
+        fig.show()
+
+
+class GaussianProbabilisticSetting(ProbabilisticDistribution):
+
+    def __init__(self):
+        pro_x = np.linspace(-5, 5, 100)
+        pro_y = []
+        for i in range(len(pro_x)):
+            pro_y.append(float(Distribution.gauss_np(pro_x[i])))
+        pro_y = np.array(pro_y)
+
+        self.prob = {"x": pro_x, "y": pro_y}
+
+        x = Symbol('x')
+        t = Symbol('t')
+
+        # integrate gaussian pdf
+        F = integrate(Distribution.gauss_sp(t), (t, -oo, x))  # gaussian cumulative density function (gaussian cdf)
+
+        dis_y = []
+        t = np.linspace(-5, 5, 100)
+        for i in range(len(t)):
+            dis_y.append(float(F.subs({x: t[i]})))
+        dis_x = pro_x
+        dis_y = np.array(dis_y)
+        self.dist = {"x": dis_x, "y": dis_y}
+
+    def save_figure(self):
+        Visualizer.visualize(x=self.prob["x"], y=self.prob["y"], title="gaussian pdf")
+        Visualizer.visualize(x=self.dist["x"], y=self.dist["y"], title="gaussian cdf")
+
+
+class BetaProbabilisticSetting(ProbabilisticDistribution):
+
+    def __init__(self):
+        pro_x = np.linspace(0, 1, 100)
+        a = 0.5
+        b = 0.5
+        pro_y = []
+        for i in range(len(pro_x)):
+            pro_y.append(float(Distribution.beta_pdf(pro_x[i], a, b)))
+        pro_y = np.array(pro_y)
+
+        self.prob = {"x": pro_x, "y": pro_y}
+
+        dis_x = pro_x
+        dis_y = stats.beta.cdf(dis_x, a, b)  # beta cumulative distribution function (beta cdf)
+        self.dist = {"x": dis_x, "y": dis_y}
+
+    def save_figure(self):
+        Visualizer.visualize(x=self.prob["x"], y=self.prob["y"], title="beta pdf")
+        Visualizer.visualize(x=self.dist["x"], y=self.dist["y"], title="beta cdf")
+
+
+class Distribution:
+
+    def gauss_np(t, mu=0, sigma=1):
```

Comment:
If you don't call these methods without initialization, define them as static method